packages/office-ui-fabric-react/src/components/TextField/TextField.base.tsx

```diff
@@ -452,31 +479,25 @@ export class TextFieldBase extends BaseComponent<ITextFieldProps, ITextFieldStat
     }
     this._latestValue = value;
 
-    this.setState(
-      {
-        value: value
-      } as ITextFieldState,
-      () => {
-        this._adjustInputHeight();
+    this.setState({ value: value } as ITextFieldState, () => {
+      this._adjustInputHeight();
 
-        if (this.props.onChange) {
-          this.props.onChange(event, value);
-        }
+      if (this.props.onChange) {
+        this.props.onChange(event, value);
+      }
 
-        if (this.props.onChanged) {
-          this.props.onChanged(value);
-        }
+      if (this.props.onChanged) {
+        this.props.onChanged(value);
       }
-    );
+    });
 
     const { validateOnFocusIn, validateOnFocusOut } = this.props;
     if (!(validateOnFocusIn || validateOnFocusOut)) {
       this._delayedValidate(value);
     }
 
-    const onBeforeChange = this.props.onBeforeChange as (newValue: any) => void;
-    onBeforeChange(value);
-  }
+    this.props.onBeforeChange!(value);
```

Comment:
This prop is optional, how could the old code not cause an issue calling it unconditionally? I feel like I'm missing something here. (Another `!` use, this one seems a bit more problematic.)

Reply:
Ahh yes, I see now that it's initialized in `defaultProps`. These use cases are why I made #6905.. but I feel like our overall convention is not to specify callbacks in `defaultProps` but rather check if they're defined and call them if so, which `TextField` varies from it seems.

Reply:
In any case I think I'd feel better if there was a check here. 😄